frontend/test/pytest/test_compiler.py

```diff
@@ -422,6 +423,17 @@ def get_output_filename(infile):
             assert observed_outfilename == expected_outfilename
             assert os.path.exists(observed_outfilename)
 
+    def test_qfunc_no_device(self):
+        """Test qfunc with device=None"""
+
+        @qjit()
+        @qfunc(1, device=None)
+        def circuit():
+            qml.PauliX(wires=0)
+            return measure(wires=0)
+
+        assert circuit()
```

Comment:
Is this something we allow? 😅
We should probably delete the `QFunc` class constructor actually, I don't think the class is intended to be used anywhere anymore, except to patch the `QNode.__call__` method.

Having said that, if we keep this for now I don't think the tests thematically fits here. `test_qnode.py` might be the closest, or simply in its own `test_qfunc.py` :)

Reply:
Hmm, that must be one of the tests I added to increase the coverage. I saw in code that we must be expecting this. Probably just removing it would be better if you are in doubts.

Reply:
> Probably just removing all this logic would be better if you are in doubts.

Exactly, I think we can remove it entirely 👍

Reply:
Done